fish-rust/src/builtins/string/length.rs

```diff
@@ -0,0 +1,78 @@
+use super::{width_without_escapes, Arguments, StringError, StringSubCommand};
+use crate::builtins::shared::{io_streams_t, STATUS_CMD_ERROR, STATUS_CMD_OK};
+use crate::ffi::parser_t;
+use crate::wchar::{wstr, WString, L};
+use crate::wchar_ext::ToWString;
+use crate::wcstringutil::split_string;
+use crate::wgetopt::{wopt, woption, woption_argument_t::*};
+
+#[derive(Default)]
+pub struct Length {
+    quiet: bool,
+    visible: bool,
+}
+
+impl StringSubCommand<'_> for Length {
+    const LONG_OPTIONS: &'static [woption<'static>] = &[
+        wopt(L!("quiet"), no_argument, 'q'),
+        wopt(L!("visible"), no_argument, 'V'),
+    ];
+    const SHORT_OPTIONS: &'static wstr = L!(":qV");
+
+    fn parse_opt(&mut self, _optarg: Option<&wstr>, c: char) -> Result<(), StringError> {
+        match c {
+            'q' => self.quiet = true,
+            'V' => self.visible = true,
+            _ => return Err(StringError::UnknownOption),
+        }
+        return Ok(());
+    }
+
+    fn handle(
+        &mut self,
+        _parser: &mut parser_t,
+        streams: &mut io_streams_t,
+        optind: &mut usize,
+        args: &[&wstr],
+    ) -> Option<libc::c_int> {
+        let mut nnonempty = 0usize;
+
+        for (arg, _) in Arguments::new(args, optind, streams, true) {
+            if self.visible {
+                // Visible length only makes sense line-wise.
+                for line in split_string(&arg, '\n') {
+                    let mut max = 0;
+                    // Carriage-return returns us to the beginning. The longest substring without
+                    // carriage-return determines the overall width.
+                    for reset in split_string(&line, '\r') {
+                        let n = width_without_escapes(&reset, 0);
+                        max = usize::max(max, n);
+                    }
+                    if max > 0 {
+                        nnonempty += 1;
+                    }
+                    if !self.quiet {
+                        streams.out.append(max.to_wstring() + L!("\n"));
+                    } else if nnonempty > 0 {
+                        return STATUS_CMD_OK;
+                    }
+                }
+            } else {
+                let n = arg.len();
+                if n > 0 {
+                    nnonempty += 1;
+                }
+                if !self.quiet {
+                    streams.out.append(WString::from(n.to_string()) + L!("\n"));
```

Comment:
I think we have `n.to_wstring()` for this

Reply:
I have it like 8 lines or so above, just forgot this one, will fix.

Reply:
```suggestion
                    streams.out.append(n.to_wstring() + L!("\n"));
```